cmd/helm/init.go

```diff
@@ -323,6 +324,13 @@ func (i *initCmd) run() error {
 		fmt.Fprintln(i.out, "Not installing Tiller due to 'client-only' flag having been set")
 	}
 
+	if version.BuildMetadata == "unreleased" && !i.opts.UseCanary && len(i.opts.ImageSpec) == 0 {
```

Comment:
This should probably check if the client has requested for `--client-only` as well, as that would affect some of the branch logic here. For example, a user running `helm init --client-only` won't need the `--canary-image` flag, so the warning should not be displayed.